components/AppBar/examples/AppBarWrappingUITableViewControllerExample.m

```diff
@@ -0,0 +1,101 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "MaterialAppBar.h"
+#import "MaterialAppBar+ColorThemer.h"
+#import "MaterialAppBar+TypographyThemer.h"
+
+@interface AppBarWrappingUITableViewControllerExample : UIViewController <UITableViewDataSource>
+
+@property(nonatomic, strong) MDCAppBarContainerViewController *appBarContainerViewController;
+@property(nonatomic, strong) MDCSemanticColorScheme *colorScheme;
+@property(nonatomic, strong) MDCTypographyScheme *typographyScheme;
+
+@end
+
+@implementation AppBarWrappingUITableViewControllerExample
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  UITableViewController *tableViewController =
+      [[UITableViewController alloc] initWithStyle:UITableViewStylePlain];
+  tableViewController.title = @"Wrapped table view";
+  self.appBarContainerViewController =
+      [[MDCAppBarContainerViewController alloc] initWithContentViewController:tableViewController];
+
+  tableViewController.tableView.dataSource = self;
+  [tableViewController.tableView registerClass:[UITableViewCell class]
+                        forCellReuseIdentifier:@"cell"];
+
+  self.appBarContainerViewController.appBar.headerViewController.headerView.trackingScrollView =
```

Comment:
oh dear, is this our current api for setting the `trackingScrollView`? Are we considering a more user-friendly approach?

Reply:
There may be some value in bubbling the flexible header APIs up to the app bar, but I'm not sure this benefit will outweigh the long-term maintenance cost (e.g. if we want to deprecate a flexible header API, we'd then have to deprecate it in two places).

Reply:
This is up to you on what you think is best. If you see changing it as more trouble than worth, then it is really important that this be clearly documented (if it's a common use case), so the user won't get lost in trying to find a setting by diving into deep level classes. I would personally try to think if there is a way to make it more intuitive for the user.

Reply:
I think https://github.com/material-components/material-components-ios/issues/348 documents a more impactful effort to improve on this API and can argue that it would be a pretty good quality-of-life improvement for our clients if we could do some level of that.

Reply:
+ https://github.com/material-components/material-components-ios/issues/347 as well.

Reply:
The main concern I have is that it's not just setting the tracking scroll view that we'd have to bubble up to the app bar - it's also the scroll view event forwarding + related APIs (of which there are many). I should be doing that in this PR anyway and will actually do right now.

Reply:
thanks for clarifying!